### PR TITLE
restrict per experiment run artifacts for trial

### PR DESCRIPTION
**Pull request by anandjakhaniya**
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-7372

### CODE REVIEW

**Review by ravishetye — COMMENTED** [review]
~this will not work  since user will end up deleting and creating artifacts~
no longer valid

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 1]
```diff
@@ -1187,6 +1191,17 @@ public void logArtifacts(String experimentRunId, List<Artifact> newArtifacts)
     }
   }
 
+  public void validateMaxArtifactsForTrial(int newArtifactsSize, int existingArtifactsSize)
+      throws ModelDBException {
+    if (app.getTrialEnabled()) {
+      if (existingArtifactsSize + newArtifactsSize > app.getMaxArtifactPerRun()) {
+        throw new ModelDBException(
+            "Maximum " + app.getMaxArtifactPerRun() + " artifacts are allow in the experimentRun",
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: "LIMIT_RUN_ARTIFACT_NUMBER : Maximum " + app.getMaxArtifactPerRun() + " artifacts are allow in the experimentRun",
this should look like above

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: Added

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 2]
```diff
@@ -1187,6 +1193,21 @@ public void logArtifacts(String experimentRunId, List<Artifact> newArtifacts)
...
+  public void validateMaxArtifactsForTrial(int newArtifactsSize, int existingArtifactsSize)
+      throws ModelDBException {
+    if (app.getTrialEnabled()) {
+      if (app.getMaxArtifactPerRun() != null
+          && existingArtifactsSize + newArtifactsSize > app.getMaxArtifactPerRun()) {
+        throw new ModelDBException(
+            ModelDBConstants.LIMIT_RUN_ARTIFACT_NUMBER
+                + "Maximum "
+                + app.getMaxArtifactPerRun()
+                + " artifacts are allow in the experimentRun",
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]:
```suggestion
                + " artifacts are allowed in an experiment run",
```

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 3]
```diff
@@ -3004,6 +3026,37 @@ public ExperimentRun cloneExperimentRun(CloneExperimentRun cloneExperimentRun, U
...
+                .build();
+        ExperimentRunPaginationDTO paginationDTO =
+            findExperimentRuns(projectDAO, userInfo, findExperimentRuns);
+        if (app.getMaxExperimentRunPerWorkspace() != null
+            && paginationDTO.getTotalRecords() >= app.getMaxExperimentRunPerWorkspace()) {
+          throw new ModelDBException(
+              ModelDBConstants.LIMIT_RUN_NUMBER
+                  + "Maximum "
+                  + app.getMaxExperimentRunPerWorkspace()
+                  + " experimentRuns are allow in the same workspace",
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]:
```suggestion
                  + " experiment runs are allowed in the same workspace",
```

**Review by ravishetye — APPROVED** [review]

**Comment by anandjakhaniya** [comment]
Created subsequent PRs for this https://github.com/VertaAI/modeldb/pull/1629 so closing this PR.

**Pull request closed without merging**